File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s936984238_comments_removed.py
def main():
    while read():
        solve()

def read():
    global number, score
    number = int(input())
    if number == 0:
        return False
    score = list(map(int, input().split()))
    return True

def solve():
    score.sort()
    total_sum = sum(score[1:number-1])
    print(total_sum // (number - 2))

main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Go/s936984238_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var (
	sc    = bufio.NewScanner(os.Stdin)
	score []int
)

func main() {
	for read() {
		solve()
	}
}

func read() bool {
	sc.Scan()
	line := sc.Text()
	number, err := strconv.Atoi(line)
	if err != nil || number == 0 {
		return false
	}

	score = make([]int, number)
	for i := 0; i < number; i++ {
		sc.Scan()
		score[i], _ = strconv.Atoi(sc.Text())
	}
	return true
}

func solve() {
	sort.Ints(score)
	sum := 0
	for i := 1; i < len(score)-1; i++ {
		sum += score[i]
	}
	fmt.Println(sum / (len(score) - 2))
}

// End of Code
